### ISSUE TITLE
Feature request: richer integration with git features.

### ISSUE THREAD

**Issue by bvandevliet** [raymon-sun/vscode-git-history#2]
**This extension is very neat!** However, a major shortcoming is the lack of proper git integration.

I believe this extension has great potential if it integrates with git features like **push**, **checkout**, **merge**, etc.
How I'd love to see this is to have below options added to the context menu on a right click, when applicable.
I mean, the current **Cut**, **Copy**, **Paste** is just not sufficient, I agree **Copy** is nice, but **Cut** and **Paste**?

### On a commit
* Checkout
* Add tag ...
* Create branch ...
* Merge into current branch ...
* Cherry pick ...
* Revert ...
* Reset ...
* Rebase on this commit ...

### On a tag
* Push tag ...
* Delete tag ...

### On a branch
* Checkout branch ...
* Push branch ...
* Rename branch ...
* Delete branch ...
* Pull into current branch ...
* Merge into current branch ...

**Comment by hansu** [comment]
I agree. At least some basic commands like checkout commit or branch would be quite useful.

> I mean, the current Cut, Copy, Paste is just not sufficient, I agree Copy is nice, but Cut and Paste?

Yes cut and Paste does not makes any sense here. 
However this context menu should be removed as it does not have any function. Even "Copy" does not work via the context menu.
